client/src/redux/actions/twitterActions.js

```diff
@@ -292,25 +286,24 @@ export const loadMoreHomeTimelineStream = ({ id, maxId }) => async (
 
 export const loadUserStreams = ({ streams }) => async (dispatch) => {
   try {
-    // create copy of current streams stored in redux
-    // remove isLoading and tweets from streams parameter
-    // compare redux stored streams with streams stored in backend
-    // if they dont match, add add the new stream to redux
-    // this way, only at first load (before streams was set up in redux) we will load all Stream components
-
-    const streamsArray = store
-      .getState()
-      .twitterReducer.streams.map((stream) => ({
-        id: stream.id,
-        type: stream.type,
-        search: stream.search ? stream.search : undefined,
-        isLoading: undefined,
-        tweets: undefined,
-        metadata: undefined,
-      }));
-    if (JSON.stringify(streamsArray) !== JSON.stringify(streams)) {
+    const streamsArray = store.getState().twitterReducer.streams;
+    let synced = true;
+    streams.map(async (stream, idx) => {
+      if (stream.id !== streamsArray[idx]) {
+        synced = false;
+      }
+    });
+    if (!synced) {
       await dispatch(updateStreams({ streams: [] }));
-      dispatch(reorderStreams({ streams }));
+      await dispatch(clearStreamsById());
+      streams.map((stream) => {
```

Comment:
There are too many actions dispatched here. It works for now because the number of streams is small, but ideally you would want to do all the processing / object construction via `map()` and then dispatch once using `updateStreams()`.

Reply:
see comment below